packages/x-components/src/x-modules/facets-next/entities/hierarchical-filter.entity.ts

```diff
@@ -42,8 +42,10 @@ export class HierarchicalFilterEntity implements FilterEntity {
     if (filter.children) {
       filter.children.forEach(childId => {
         const child = this.getFilterById(childId);
-        this.saveFilter({ ...child, selected: false });
-        this.deselectDescendants(child);
+        if (child) {
```

Comment:
There is a problem with the current approach, and it is that a filter child might not be (yet) in the store. We could maybe add them always before using an entity (at the service), or pass to the entities the list of filters they should work with so they do not have to rely on querying the store.

Reply:
it is ok as it is at the moment.